azext_iot/operations/hub.py

```diff
@@ -2187,6 +2189,11 @@ def iot_simulate_device(
     )
     token = None
 
+    if method_response_payload:
```

Comment:
We should throw an error if custom values are provided for `method_response_code` and `method_response_payload` when simulating via http so users aren't confused. Similar to what we have when trying to change c2d receive settlement when we are simulating an mqtt protocol device.

Reply:
And a test for this.

Reply:
I think instead of checking for http and failing we should check if the protocol is not mqtt - and then fail if custom values are provided for method_response_code and method_response_payload. That makes us future proof.